src/main/java/com/fp3/haras/view/screens/stable/StableCreate.java

```diff
@@ -390,24 +554,157 @@ private void btnCancelActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIR
+                } else {
+                    Servico prod = (Servico) consumoGeral.get(boxConsumo.getSelectedIndex());
+                    
+                    for (int i = 0; i < novoConsumo.size(); i++) {
+                        Consumo c = novoConsumo.get(i);
+                        if (c.getServico() != null && c.getServico().equals(prod)) {
+                            alreadyExists = true;
+                            existsIndex = i;
+                            break;
+                        }
+                    }
+                    
+                    if (!alreadyExists && (int)spnProdUsed.getValue() > 0) {
+                        novoConsumo.add(new Consumo(novaEstadia, prod, (int)spnProdUsed.getValue()));
+                        updateServico(prod);
+                    
+                    } else if (alreadyExists && (int)spnProdUsed.getValue() != 0){
+                        novoConsumo.get(existsIndex).setQtd((int)spnProdUsed.getValue());
+                        updateServico(prod);
+                    
+                    } else if (alreadyExists && (int)spnProdUsed.getValue() == 0) {
+                        novoConsumo.remove(existsIndex);
+                        updateServico(prod);
+                    
+                    } else if (!alreadyExists && (int)spnProdUsed.getValue() == 0) {
+                        JOptionPane.showMessageDialog(null, "Use '0' apenas para apagar itens ativos", null, JOptionPane.WARNING_MESSAGE);
+                    }
+                }
+            }
+        } else {
+            JOptionPane.showMessageDialog(null, "Selecione o tipo da estadia!", null, JOptionPane.ERROR_MESSAGE, null);
+        }
     }//GEN-LAST:event_btnApplyActionPerformed
 
+    private void boxConsumoActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_boxConsumoActionPerformed
+        if (boxConsumo.getSelectedItem() != "" && boxConsumo.getSelectedIndex() > 0) {
+            spnProdUsed.setEnabled(true);
+            double total;
+            if (consumoGeral.get(boxConsumo.getSelectedIndex()).getClass() == Produto.class) {
+                Produto prod = (Produto) consumoGeral.get(boxConsumo.getSelectedIndex());
+                txtIndividualValue.setText(String.valueOf(prod.getPdv()));
+                total = prod.getPdv()*(int)spnProdUsed.getValue();
+                txtFullValue.setText(String.valueOf(total));
+                spnProdUsed.setValue(1);
+            } else if (consumoGeral.get(boxConsumo.getSelectedIndex()).getClass() == Servico.class){
+                Servico prod = (Servico) consumoGeral.get(boxConsumo.getSelectedIndex());
+                txtIndividualValue.setText(String.valueOf(prod.getPrice()));
+                total = prod.getPrice()*(int)spnProdUsed.getValue();
+                txtFullValue.setText(String.valueOf(total));
+                spnProdUsed.setValue(1);
+            } else {
+                System.err.println("IMPOSSIVEL DEFINIR TIPO DE CONSUMO");
```

Comment:
I'm not sure if I understand all this code, but I think here would be better to display a message to the user with a JOptionPane